### ISSUE TITLE
'All Movies' and 'All TV Shows' folders should reflect Netflix genres

### ISSUE THREAD

**Issue by lekma** [CastagnaIT/plugin.video.netflix#48]
*I'm submitting a ...*
  - [ ] bug report
  - [x] feature request
  - [ ] support request

### Description

The content of the folders 'All Movies' and 'All TV Shows' seems inconsistent/random, it often does not include major genres (like Sci-Fi) and rarely reflects the genres offered on Netflix's website (as seen below).
I believe we would have better accessibility to the catalog if those folders were (at least) coherent with the ones from the site.

![netflix-series-genres](https://user-images.githubusercontent.com/1482125/55825910-2b3fa200-5b07-11e9-9573-12c998b68236.png)

![netflix-films-genres](https://user-images.githubusercontent.com/1482125/55826089-7e195980-5b07-11e9-8a8f-4d3bf8ae6958.png)

**Expected behavior:** [What you expected to happen]

![kodi-series-subgenres](https://user-images.githubusercontent.com/1482125/55824559-5bd20c80-5b04-11e9-8842-f15f6cc26957.png)

![kodi-films-subgenres](https://user-images.githubusercontent.com/1482125/55824632-8623ca00-5b04-11e9-8c0b-f5cdf65c2db1.png)

**Actual behavior:** [What actually happened]

![kodi-series-genres](https://user-images.githubusercontent.com/1482125/55824688-a94e7980-5b04-11e9-87f3-054da075a083.png)

Note the difference in numbers of genres: 12 instead of 17.

![kodi-films-genres](https://user-images.githubusercontent.com/1482125/55824733-c2efc100-5b04-11e9-9846-c196f65fb6f1.png)

Same here: 13 genres instead of 18.

### Addon version used

0.14.0~beta23

### Installation

* [x] I'm using this Netflix Repo
* [ ] I'm using other Netflix Repo (Please specify which)
* [ ] I'm using a different source (Please tell which)

## Other information

@CastagnaIT: to produce the mockup screenshots I changed the query currently used to produce the list, you can have a look here: [_get_subgenres()](https://github.com/lekma/plugin.video.netflix/blob/genres/resources/lib/api/genres.py#L130). It may be as simple as querying the correct lolomo, but I couldn't find it (lack of documentation on the shakti api...).

**Comment by jakermx** [comment]
I just install your fork, nice work, but  I will be nice to avoid hardcoding, I have a Mexico Profile  and It shows British as Genre, It should show Mexican and Latin American.

I think here
plot_fmt = common.get_local_string(30087)

And would be nice change this
path = [['genres', uid, 'subgenres', {'from': 0, 'to': 47}, 'id']]

And do something automatic, and catch the exception when fail, so it stops at that point...

**Comment by CastagnaIT** [comment]
I think you're making a lot of confusion...
-nothing of this, is hardcoded

-get_local_string() read the language on resource strings, based on skin language of kodi, if you find any words in English it is possible that your translation is missing in the resource files
check "/resources/language/resource.language.[your language code]/strings.po
compare with the default "/resources/language/resource.language.en_gb/strings.po"

-the modification proposed to the menu string doesn't make sense

-the subgenres totally not implemented, the code that you see has only partially begun


for the missing genres, i can verify the responses from the server over the weekend, and I'll let you know

but do not think that everything you see on the website must respect 100% of the addon menus
the website does many things and changes often, it is already very much to keep it all working

**Comment by jakermx** [comment]
Maybe I did not express myself in the way I wanted, what I meant is that somehow when using the AddOn, I got the British Genre on Movies or TV Shows, instead of Mexican and Latin, whom are the ones I get while browsing on my computer or Netflix App, since I am not know about Navigation on Netflix, and since when I buikld Dynamic Menus, on other plattforms than Netflix , I use some queries to get Menus and Permissions thats why I though it was hard coded, based on your response and some quick investigation, please apologize me, it was my fault,  so  I will said that It would be very cool to have a Setting Screen for Building Menu Items, and advance tab for editing genres and sub genres...but since I  know, the whole idea of this AddOn is to make it work like the browser page, this would be a Plugin for this AddOn.

Sorry for the misunderstanding... nice work!!

**Comment by CastagnaIT** [comment]
for the gender list, our addon currently requests the list you can see on the homepage that's why you can't find a match,
to implement that kind of list you want, it's not immediate, first of all because at the moment I haven't found the query to request that list yet, secondly there is also to separate the genres for TV series and movies.

for the language you have to set your language in the netflix panel of the site, in the profile language, so the results you find in netflix addon will be returned in your language

**Comment by lekma** [comment]
I think I need to clarify the intent of this feature request.

Yesterday's 'All Movies':
- 'Independent Dramas' (intersection of 'Independent' and 'Dramas'?)
- 'Comedies'
- 'Action & Adventure'
- 'Thrillers'
- 'Dramas'
- 'Children & Family Films'
- 'Action Thrillers' (intersection of 'Action' and 'Thrillers'?)
- 'Documentaries'
- 'Romantic Films'
- 'Crime Films'
- 'International Films' (aren't all movies "international", what does that even mean??)
- 'Horror Films'
- 'Films for ages 11 to 12' (subset of 'Kids & Family')?

Missing (main) genres: 'Independent', 'Sci-Fi'.

Today's 'All Movies':
- 'Oscar-winning Films' (are oscars still a thing?)
- 'Comedies'
- 'Action & Adventure'
- 'Thrillers'
- 'Children & Family Films'
- 'Crime Films'
- 'Romantic Films'
- 'Documentaries'
- 'Dramas'
- 'Films for ages 11 to 12' (subset of 'Kids & Family')?
- 'Action Thrillers' (intersection of 'Action' and 'Thrillers'?)
- 'Films for ages 2 to 4' (wait, what? (subset of 'Kids & Family'?))
- 'International Films' (??)

Missing: 'Horror', 'Independent', 'Sci-Fi'.

And those were good days compared to some in the past.

So, the point I'm trying to make is _not_ that we should exactly replicate the genres menu from Netflix's website but rather that maybe we could provide a base list that is more broad/complete and especially more consistent than the current one (maybe _inspired_ by the one from the site).

The [genres branch](https://github.com/lekma/plugin.video.netflix/tree/genres) should not be taken as a proposal for a technical solution. It is here only to provide a mockup implementation meant to illustrate the _possibility_ of a solution (and for me to learn about kodi).

I wish more users would weigh in on the merit of this proposal (maybe I'm completely wrong and people are happy with what is offered right now).

TLDR: maybe we should rename 'All Movies' to 'Some Movies' and 'All TV Shows to 'Some TV Shows' :smile:.

**Comment by CastagnaIT** [comment]
i have understood the problem well, but you have not understood the procedure to get the list of genres on the addon.

the comparison with the first or second screenshots posted at the beginning is totally wrong,
the list of genres shown in those two screenshots refer to the netflix website menu, who gets that list in a different way i haven't been able to understand how

the list of genres that we have comes from lolomo is the list displayed on the netflix homepage but filtered by genre

so the comparison you're making is completely wrong, every time netflix change the elements of the homepage (consequently the lolomo contents), the lists contained in our genre menu will be changed respectively...

**Comment by lekma** [comment]
> the comparison with the first or second screenshots posted at the beginning is totally wrong, the list of genres shown in those two screenshots refer to the netflix website menu, who gets that list in a different way i haven't been able to understand how

And I've tried to show you there is a (relatively) easy way to replicate that (which  unfortunately doesn't involve a lolomo, maybe it's time we forgo those for the genres lists?).

> the list of genres that we have comes from lolomo is the list displayed on the netflix homepage but filtered by genre

nope it isn't, please have a [look](https://github.com/CastagnaIT/plugin.video.netflix/blob/master/resources/lib/api/shakti.py#L157), what is selected comes from the 'rw' field (which I believe refers to 'related something') the 'subgenres' part at the end of the query isn't even processed by the Lolomo object (I think?).

Of course, you know the code and the backend better than I do and I might be completely wrong here, but as shown [here](https://github.com/lekma/plugin.video.netflix/blob/genres/resources/lib/api/genres.py) we can query shakti and have a result similar to the one from the website (without any specific processing).

Again, I'm not hung up on a technical solution right now (I have one for the time being) . I'm more interested in users opinion on the merit of having a consistent/reliable genres list for movies and tv shows.

You/me/others can work on an actual proposal if there is interest.

**Comment by CastagnaIT** [comment]
The word "rw" should be the type of order applied to the results as an example "su" or "az",
but the code mentioned is to get the list of contents of a selected genre, not the list of actual genres......

**Comment by lekma** [comment]
> The word "rw" should be the type of order applied to the results as an example "su" or "az",

well, I don't believe so

> but the code mentioned is to get the list of contents of a selected genre, not the list of actual genres......

yet it is the code used to generate the lists we are talking about...

**Comment by CastagnaIT** [comment]
so as you think.. if you want to perform experiments you are free to do them, i do not waste time. However subgenres are: subgenres of genres

**Comment by lekma** [comment]
>  However subgenres are: subgenres of genres

and 'Movies' and 'TV Shows' are treated as genres by shakti...

**Comment by jakermx** [comment]
It is easy... anything from main page is Genre and anything that is shown after select a Genre is Sub genre, since I know you have the "secret" genre list, anything is the treated equal, but for functional pro purses there are genres and sub-ones

**Comment by CastagnaIT** [comment]
genres/sub-genres will be available in the next release
